tests/sentry/auth_v2/endpoints/test_auth_merge_user_accounts.py

```diff
@@ -0,0 +1,130 @@
+        self.create_user(username="garden variety mifu", email="mifu@example.com")
+
+        org1 = self.create_organization(name="hojicha")
+        org2 = self.create_organization(name="matcha")
+        org3 = self.create_organization(name="oolong")
+
+        self.create_member(user=user1, organization=org1)
+        self.create_member(user=user1, organization=org2)
+        self.create_member(user=user2, organization=org3)
+
+        self.login_as(user1)
+        response = self.get_success_response()
+
+        assert response.data[0]["organizations"] == [org1.name, org2.name]
+        assert response.data[1]["organizations"] == [org3.name]
+        assert response.data[2]["organizations"] == []
+
+
+@control_silo_test
+class MergeUserAccountsWithSharedEmailTest(APITestCase):
+    endpoint = "sentry-api-0-auth-merge-accounts"
+    method = "post"
+
+    def setUp(self):
+        self.user1 = self.create_user(username="powerful mifu", email="mifu@email.com")
+        self.user2 = self.create_user(username="transcendent mifu", email="mifu@email.com")
+        self.user3 = self.create_user(username="garden variety mifu", email="mifu@email.com")
+
+        org1 = self.create_organization(name="hojicha")
+        org2 = self.create_organization(name="matcha")
+        org3 = self.create_organization(name="oolong")
+
+        self.create_member(user=self.user1, organization=org1)
+        self.create_member(user=self.user1, organization=org2)
+        self.create_member(user=self.user2, organization=org3)
+
+        self.unrelated_user = self.create_user(email="foo@bar.com")
+
+        self.login_as(self.user1)
+
+        self.verification_code = UserMergeVerificationCode.objects.create(user_id=self.user1.id)
+
+    def test_simple(self):
+        data = {
+            "ids_to_merge": [self.user2.id],
+            "verification_code": self.verification_code.token,
+        }
+        self.get_success_response(**data)
+
+        assert not User.objects.filter(id=self.user2.id).exists()
+        assert not User.objects.filter(id=self.user3.id).exists()
+
+    def test_incorrect_code(self):
+        data = {
+            "ids_to_merge": [self.user2.id],
+            "verification_code": "hello",
+        }
+        response = self.get_error_response(**data)
+        assert response.status_code == 403
+        assert response.data == {"error": "Incorrect verification code"}
+
+    def test_primary_email_unverified(self):
+        user_email = UserEmail.objects.get(user_id=self.user1.id, email="mifu@email.com")
+        user_email.update(is_verified=False)
+        data = {
+            "ids_to_merge": [self.user2.id],
+            "verification_code": self.verification_code.token,
+        }
+        response = self.get_error_response(**data)
+        assert response.status_code == 403
+        assert response.data == {
+            "error": "You must verify your primary email address to use this endpoint"
+        }
```

Comment:
To confirm, we want to force them through a separate email verification flow even if they provide a valid verification code here?